instrumentation-tools/src/api_surface.py

```diff
@@ -0,0 +1,76 @@
+#  Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+#  This product includes software developed at Datadog (https://www.datadoghq.com/).
+#  Copyright 2016-Present Datadog, Inc.
+
+import re
+from math import floor
+
+DIVIDER = "#"
+TESTABLE_API_REG_EX = re.compile("^(.+)(constructor|fun)(.+)")
+ROOT_REG_EX = re.compile("^(?<!companion)(\\s*)[a-zA-z]*(class|interface|enum|object)(.*)(\\.|\\s)(.+)$")
+
+"""
+In this function we will sanitize the current API prefix as following.
+Say we have the following apiSurface file:
+
+class Foo
+  fun a
+  class Bar
+    fun b
+class AnotherFoo
+  fun c
+
+When we reach line `class AnotherFoo` the prefix will be `Foo#Bar` but the API key 
+for `fun c` should be: `AnotherFoo#fun c` therefore based on the indentation difference (2) 
+in this case we will substract 2 entities from the prefix (Foo and Bar) which will result in an empty prefix at
+this point. Going further we will attach the current entity `AnotherFoo#`
+
+"""
+def sanitize_prefix(previous_indent: int,
+                    current_indent: int,
+                    prefix: str) -> str:
+    # we divide the difference to 2 as the indentation step in apiSurface tool is 2
+    difference = floor((current_indent - previous_indent) / 2)
```

Comment:
Maybe you could have a constant `INDENT_SIZE=2` instead of using a magic number ?